prisma/schema.prisma

```diff
@@ -99,51 +105,143 @@ model EmailVerificationCode {
+  campaignId    String         @unique
+  AudienceUsers AudienceUser[]
+  createdAt     DateTime       @default(now())
+  updatedAt     DateTime       @updatedAt
+}
+
+model Audiences {
+  id           String         @id @default(cuid())
+  name         String
+  slug         String
+  Campaigns    Campaign[]
+  AudienceUser AudienceUser[]
+  App          App            @relation(fields: [appId], references: [id])
+  appId        String
+  public       Boolean        @default(false)
+  createdAt    DateTime       @default(now())
+  updatedAt    DateTime?      @updatedAt
+  deletedAt    DateTime?
+}
+
+model AudienceUser {
+  id                    String                 @id @default(cuid())
+  Audiences             Audiences[]
+  CampaignDistribution  CampaignDistribution[]
+  User                  User?                  @relation(fields: [userId], references: [id])
+  userId                String?                @unique
+  externalWalletAddress String?                @unique
+  createdAt             DateTime               @default(now())
+  updatedAt             DateTime?              @updatedAt
+  deletedAt             DateTime?
+}
+
+model Ticket {
+  id                String              @id @default(cuid())
+  address           String              @unique
+  name              String
+  slug              String
+  DevelopersAccount DeveloperAccount?   @relation(fields: [developerId], references: [id])
+  developerId       String?
+  metadataUrl       String?
+  metadataPayload   Json
+  App               App                 @relation(fields: [appId], references: [id])
+  appId             String
+  Event             Event               @relation(fields: [eventId], references: [id])
+  eventId           String
+  TicketInteraction TicketInteraction[]
+  Entrance          Entrance?
+  Campaigns         Campaign[]
+  createdAt         DateTime            @default(now())
+  updatedAt         DateTime            @updatedAt
+
+  @@unique([slug, eventId])
+}
+
+model Entrance {
+  id                  String                @id @default(cuid())
+  address             String                @unique
+  name                String
+  slug                String
+  DevelopersAccount   DeveloperAccount?     @relation(fields: [developerId], references: [id])
+  developerId         String?
+  metadataUrl         String?
+  metadataPayload     Json
+  App                 App                   @relation(fields: [appId], references: [id])
+  appId               String
+  Event               Event                 @relation(fields: [eventId], references: [id])
+  eventId             String
+  Ticket              Ticket?               @relation(fields: [ticketId], references: [id])
+  ticketId            String?               @unique
+  EntranceInteraction EntranceInteraction[]
+  createdAt           DateTime              @default(now())
+  updatedAt           DateTime              @updatedAt
+
+  @@unique([slug, eventId])
+}
+
+model TicketInteraction {
```

Comment:
can't we have single model `Interaction`? all fields are the same except ticketId/entranceId. 
we can just include both of those fields in the Interaction model and add custom migration beacsue prisma currently does not support check validation (I will handle it, for now we can just add optional ticket and entrance. maybe in the future there will be more smart contract based models)